root/opt/puppetlabs/puppet/lib/ruby/2.1.0/facter/esmithdb.rb

```diff
@@ -7,7 +7,7 @@
     setcode do
         dbs = {}
         Dir.entries('/var/lib/nethserver/db').each do |db|
-            next if (db == '.') || (db == '..')
+            next if (db =~ /^\./)
             tmp = Facter::Core::Execution.exec("/sbin/e-smith/db #{db} printjson")
             data = JSON.parse(tmp)
```

Comment:
Shall we protect against JSON encoding errors? A begin/rescue block might help...